src/components/ModalPortal.js

```diff
@@ -285,6 +285,8 @@ export default class ModalPortal extends Component {
   };
 
   handleOverlayOnClick = event => {
+    event.stopPropagation();
```

Comment:
I came across this pull request as my events aren't propagating right now when I want them to (I'm doing something wrong in my app). My scenario is that I have a transparent overlay and `shouldCloseOnOverlayClick` is set. I want the modal to close and underlying control to be clicked.

I don't think that `event.stopPropagation()` should always happen; it should be opt-in (or a setting to control the behavior). With this change, I believe it's possible to break existing use of `react-modal` that relies on this behavior.

Reply:
Thanks for the review @smcenlly. I will take a look

Reply:
We can add the stopPropagation there...we need to be careful here to only stop if the event is target is the overlay.

Reply:
I fixed this, in this commit: https://github.com/reactjs/react-modal/pull/940/commits/4dd8ee3cf9e09606bd3170e97712447756e39480

Reply:
Muito bem...um potiguar por aqui!!

I think we are close to get this ready...can this `event` not be related to the overlay? I'm not sure if that is a valid case, but it would be nice to check it.

Reply:
This needs to be merged and released if this allows handling overlay click events.